syntax = "proto3";

package librtcdcpp;

message Callback {
  message onCandidate {
    string candidate = 1;
    string sdpMid = 2;
    uint32 sdpMLineIndex = 3;
  }
  message onMessage {
    string message = 1;
    bytes message_bytes = 2;
  }
  // For cb's without args
  enum CBSansArgs {
    NONE = 0;
    ON_CHANNEL = 1;
    ON_CLOSE = 2;
  }
  CBSansArgs cbwo_args = 1;
  // For cb's with args
  onCandidate on_cand = 2;
  onMessage on_msg = 3;
}
